internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -88,6 +89,51 @@ func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded(client cli
 	return isControlPlane && allControlPlanesMatchVersion && matchesClusterVersion, nil
 }
 
+// IsNodeUpgradeable returns whether a given node is upgradeable or not, taking global cluster restrictions into account.
+// If all preconditions are met for the given node, no error will be returned.
+func (nviu NodeVersionInfoUpdate) IsNodeUpgradeable(client clientset.Interface, currentClusterVersion *version.Version) error {
```

Comment:
from the naming of the function it implies that a bool is returned, we have that covention in some other places as well. As it's easier to just return `err` here and not a bool and an err, maybe I'd rename the function to something like `NodeUpgradeableCheck`